clients/python/dd_client.py

```diff
@@ -1,118 +1,56 @@
+# -*- coding: utf-8 -*-
 """
+@author: Evgeny BAZAROV <baz.evgenii@gmail.com>
+
+@brief:
 DeepDetect Python client
 
 Licence:
 Copyright (c) 2015 Emmanuel Benazera
 
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 
-The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
 
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
 
 """
 
-try:
-    import urllib.request as urllib2
-except ImportError:
-    import urllib2
-
-try:
-    import http.client as httplib
-except ImportError:
-    import httplib
-
-
-import os.path
-import json
-import uuid
-import datetime
-
-
-VERBOSE=False
-DD_TIMEOUT = 2000 # seconds, for long blocking training alls, as needed
-
-def LOG(msg):
-    """Output a log message."""
-    # XXX: may want to use python log manager classes instead of this stupid print
-    if VERBOSE:
-        msg = str(datetime.datetime.now()) + ' ' + msg
-        print (msg)
+import requests
 
-### Exception classes :
-
-class DDCommunicationError(Exception):
```

Comment:
This exception are handled by library **requests** :  
- Communication. It will check status code and if needed will raise exception. **[errors-and-exceptions](http://docs.python-requests.org/en/master/user/quickstart/#errors-and-exceptions)**  
- Data error. If there is no `json` object that could be decoded it will raise error. **[json-response-content](http://docs.python-requests.org/en/master/user/quickstart/#json-response-content)**